src/glue/jobs/json_to_parquet.py

```diff
@@ -182,11 +183,86 @@ def drop_deleted_healthkit_data(
     )
     return table_with_deleted_samples_removed
 
+def archive_existing_datasets(
+        glue_context: GlueContext,
+        bucket: str,
+        key_prefix: str,
+        workflow_run_id: str,
+        delete_upon_completion: bool
+    ) -> None:
+    """
+    Archives existing datasets in S3 by copying them to a timestamped subfolder
+    within an "archive" folder. The format of the timestamped subfolder is:
+
+        {year}_{month}_{day}_{workflow_run_id}
+
+    If the dataset does not exist at `key_prefix` or the dataset is empty,
+    no action will be taken.
+
+    Args:
+        glue_context (GlueContext): The Glue context
+        bucket (str): The name of the S3 bucket containing the dataset to archive.
+        key_prefix (str): The S3 key prefix of the objects to be archived.
+            Ensure that this ends with a trailing "/" to avoid matching
+            child datasets.
+        workflow_run_id (str): The Glue workflow run ID.
+        delete_upon_completion (bool): Whether to delete the dataset after archiving.
+
+    Returns:
+        None
+
+    Raises:
+        N/A
+    """
+    logger = glue_context.get_logger()
+    s3_client = boto3.client("s3")
+    response = s3_client.list_objects_v2(
+            Bucket=bucket,
+            Prefix=key_prefix
+    )
+    if "Contents" not in response:
```

Comment:
In what scenario will there not be "Contents"?

Reply:
If the key prefix does not exist or if it does exist but is empty. That would mean that this is the first time we are processing this data type (or all the data for this data type was deleted for some reason) and there is nothing to archive.